### DISCUSSION TITLE
Device Unlock

### DISCUSSION THREAD

**Discussion by trymeouteh** [keepassxreboot/keepassxc#10181] [Ideas]
The ability to save the keepass database password, keyfile, etc inside of KeepassXC in an encrypted form. And users can set a shorter password or passcode to unlock their databases.

For exmaple, lets say you have two databases. The password database 1 is "darkroom overcast manicure comprised december demise" and the password for database 2 is "quintet dwelled urologist manila smartly cyclist". Instead of having to always type these two password out to unlock the databases. The user can save these databases as remembered databases in KeePassXC and instead enter in "445889" to unlock database 1 and enter in "wild hunt pop" to unlock database 2.

This allows for easier and quicker unlock while keeping the actual keepass database secured since it will be stored with encryption, and no need to change the password of the keepass database itself to a shorter unsecure password.

Device unlock is a feature in KeePass DX.

**Comment by droidmonkey** [comment]
https://keepassxc.org/docs/KeePassXC_UserGuide#_automatic_database_opening

**Comment by trymeouteh** [comment]
> https://keepassxc.org/docs/KeePassXC_UserGuide#_automatic_database_opening

I would like to suggest device unlocking without having to create a keepass database as the master database, but have it as a built in feature to the client with a section in the app settings to manage the database password and keyfiles. This will be a more easier user experience to setup.

**Reply by droidmonkey** [reply]
That's why we have quick unlock, and soon to be remembered.
